src/api/integrations/ordnance-survey/index.ts

```diff
@@ -0,0 +1,47 @@
+import axios from 'axios';
+import dotenv from 'dotenv';
+import { OrdnanceSurveyAPIResponse } from '../../types';
+import { ORDNANCE_SURVEY_QUERY_URL } from '../../constants';
+
+dotenv.config();
+
+const { ORDNANCE_SURVEY_API_KEY, ORDNANCE_SURVEY_API_URL } = process.env;
+
+/**
+ * ordnanceSurvey
+ * makes ordnance survey API call and returns address
+ * @param {String} postcode
+ * @returns {OrdanceSurveyAPIResponse} OrdanceSurveyAPIResponse object
+ */
+const ordnanceSurvey = {
+  get: async (postcode: string): Promise<OrdnanceSurveyAPIResponse> => {
+    try {
+      const response = await axios({
+        method: 'get',
+        url: `${ORDNANCE_SURVEY_API_URL}${ORDNANCE_SURVEY_QUERY_URL}${postcode}&key=${ORDNANCE_SURVEY_API_KEY}`,
+        validateStatus(status) {
+          const acceptableStatus = [200, 404];
+          return acceptableStatus.includes(status);
+        },
+      });
+
+      // if no data in response or status is not 200 then return success as false
+      if (!response?.data?.results || response.status !== 200) {
```

Comment:
Looks like a test is missing for checking `succes=false` is returned if `.results` doesn't exist  i.e empty `data` object?